### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Cy90sure**
- [DEMO LINK](https://Cy90sure.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://Cy90sure.github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

File: src/style.css [hunk 1]
```diff
@@ -1,3 +1,133 @@
...
+  padding: 0;
+  color: white;
+  font-size: 14px;
+  font-weight: 700;
+  line-height: 40px;
+  text-decoration: none;
+  text-align: center;
+  background-color: rgba(0, 172, 220, 1);
+  border-radius: 5px;
+}
```
Comment (xiimiik) [inline src/style.css:127]: add transition for values which change

**Review by xiimiik — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 2]
```diff
@@ -5,9 +5,46 @@
...
   <body>
-    <h1>Product cards</h1>
+    <div class="card" data-qa="card">
+      <img
+        class="card__picture"
+        src="images/imac.jpeg"
+        alt="APPLE A1419 iMac"
+      >
+
+      <h1 class="card__item-name">APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</h1>
```
Comment (mykhalenych) [inline src/index.html:21]: change to h2

File: src/index.html [hunk 3]
```diff
@@ -5,9 +5,46 @@
...
+
+        <p class="card__reviewsText">Reviews: 5</p>
+      </div>
+
+      <div class="card__price">
+        <p class="card__priceText">Price:</p>
+        <p class="card__priceNumb">$2,199</p>
+      </div>
+
+      <a
```
Comment (mykhalenych) [inline src/index.html:42]:
```suggestion
      <a
        data-qa="hover"
        class="card__buy"
        href="https://www.moyo.ua/monoblok_apple_a1419_imac_27_retina_5k_mned2ua_a/361498.html"
      >
        buy
      </a>
```

File: src/style.css [hunk 4]
```diff
@@ -1,3 +1,132 @@
 body {
   margin: 0;
+  font-family: Roboto, Helvetica, sans-serif;
+  font-weight: 400;
+}
+
+.card {
+  box-sizing: border-box;
```
Comment (mykhalenych) [inline src/style.css:8]: move to `*` selector

File: src/style.css [hunk 5]
```diff
@@ -1,3 +1,132 @@
...
+  background-image: url(images/star.svg);
+  background-position: center;
+  background-repeat: no-repeat;
+}
+
+.card__star:last-child {
+  margin-right: 0;
+}
+
+.card__reviewsText {
```
Comment (mykhalenych) [inline src/style.css:80]: use kebab case for class names

**Review by mykhalenych — CHANGES_REQUESTED** [review]

File: src/index.html (hunk 3 above)
Comment (xiimiik) [reply src/index.html:42]: not fixed

File: src/style.css (hunk 4 above)
Comment (xiimiik) [reply src/style.css:11]: not fixed

File: src/style.css [hunk 6]
```diff
@@ -1,3 +1,132 @@
...
+
+.card__priceNumb {
+  margin: 0;
+  font-size: 16px;
+  font-weight: 700;
+  line-height: 18px;
+}
+
+.card__buy {
+  box-sizing: border-box;
```
[the remark this answers is not in the document]
Comment (xiimiik) [reply src/style.css:113]: not fixed

**Review by xiimiik — CHANGES_REQUESTED** [review]
Wouldn't it be better to set padding for the card?
and remove these strange horizontal margins, they should be ordered, that is, if you use `margin-top` for indentation between elements, then only it should be used, no `margin-bottom` together with `margin-top`
image
image

File: src/style.css [hunk 7]
```diff
@@ -1,3 +1,139 @@
...
+  margin-top: 40px;
+}
+
+.card__code {
+  margin: 0;
+  margin-top: 4px;
+  font-size: 10px;
+  font-weight: 400;
+  line-height: 14px;
+  text-align: left;
```
Comment (Viktor-Kost) [inline src/style.css:48]: it' by default as well
```suggestion
```

File: src/style.css [hunk 8]
```diff
@@ -1,3 +1,139 @@
...
+  color: var(--secondary-text-color);
+  align-self: flex-start;
+}
+
+.card__reviews {
+  margin: 0;
```
Comment (Viktor-Kost) [inline src/style.css:54]: just reset default margin for `h2` and `p` tags above
```suggestion
```

File: src/style.css [hunk 9]
```diff
@@ -1,3 +1,139 @@
...
+  margin-top: 16px;
+  display: flex;
+  width: 100%;
+  justify-content: space-between;
+}
+
+.card__starbar {
+  display: flex;
+  justify-self: left;
```
Comment (Viktor-Kost) [inline src/style.css:63]: redundant
```suggestion
```

File: src/style.css [hunk 10]
```diff
@@ -1,3 +1,139 @@
...
+.card__starbar--3 :nth-child(-n+3),
+.card__starbar--4 :nth-child(-n+4),
+.card__starbar--5 :nth-child(-n+5) {
+  background-image: url(images/star-active.svg);
+}
+
+.card__star {
+  height: 16px;
+  width: 16px;
+  margin-right: 4px;
```
Comment (Viktor-Kost) [inline src/style.css:78]: we will use`:not(:last-child)`
```suggestion
```

File: src/style.css [hunk 11]
```diff
@@ -1,3 +1,139 @@
...
+  background-image: url(images/star.svg);
+  background-position: center;
+  background-repeat: no-repeat;
+}
+
+.card__star:last-child {
+  margin-right: 0;
+}
```
Comment (Viktor-Kost) [inline src/style.css:86]:
```suggestion
.card__star:not(:last-child) {
  margin-right: 4px;
}
```

File: src/style.css [hunk 12]
```diff
@@ -1,3 +1,139 @@
...
+
+.card__reviews-text {
+  font-size: 10px;
+  font-weight: 400;
+  line-height: 16px;
+  margin: 0;
```
Comment (Viktor-Kost) [inline src/style.css:92]: just reset default margin for `h2` and `p` tags above
```suggestion
```

File: src/style.css [hunk 13]
```diff
@@ -1,3 +1,139 @@
...
+  background-repeat: no-repeat;
+}
+
+.card__star:last-child {
+  margin-right: 0;
+}
+
+.card__reviews-text {
+  font-size: 10px;
+  font-weight: 400;
```
Comment (Viktor-Kost) [inline src/style.css:90]: default
```suggestion;
```

File: src/style.css [hunk 14]
```diff
@@ -1,3 +1,139 @@
...
+
+.card__price {
+  width: 100%;
+  display: flex;
+  margin-top: 24px;
+  justify-content: space-between;
+}
+
+.card__price-text {
+  margin: 0;
```
Comment (Viktor-Kost) [inline src/style.css:103]: just reset default margin for `h2` and `p` tags above
```suggestion
```

File: src/style.css [hunk 15]
```diff
@@ -1,3 +1,139 @@
...
+  font-size: 12px;
+  font-weight: 400;
```
Comment (Viktor-Kost) [inline src/style.css:105]: default
```suggestion
```

File: src/style.css [hunk 16]
```diff
@@ -1,3 +1,139 @@
...
+  line-height: 18px;
+  color: var(--secondary-text-color);
+}
+
+.card__price-numb {
+  margin: 0;
```
Comment (Viktor-Kost) [inline src/style.css:111]: just reset default margin for `h2` and `p` tags above
```suggestion
```

File: src/style.css [hunk 17]
```diff
@@ -1,3 +1,139 @@
...
+  font-size: 16px;
+  font-weight: 700;
+  line-height: 18px;
+}
+
+.card__buy {
+  margin-top: 16px;
+  width: 100%;
+  height: 40px;
+  padding: 0;
```
Comment (Viktor-Kost) [inline src/style.css:121]: it's better not hardcode height for elements with content inside. Use padding
```suggestion
  padding: 11px;
```

File: src/style.css [hunk 18]
```diff
@@ -1,3 +1,139 @@
...
+  color: white;
+  font-size: 14px;
+  font-weight: 700;
+  line-height: 40px;
```
Comment (Viktor-Kost) [inline src/style.css:125]: such a big value
```suggestion
  line-height: 16px;
```

File: src/style.css [hunk 19]
```diff
@@ -1,3 +1,139 @@
...
+  text-decoration: none;
+  text-align: center;
+  background-color: var(--button-color);
+  border: 1px solid var(--button-color);
+  border-radius: 5px;
+  text-transform: uppercase;
+  transition: all 500ms;
```
Comment (Viktor-Kost) [inline src/style.css:132]: specify what exactly you'd like to change
```suggestion
  transition: color 500ms, border-color 500ms, background-color 500ms;
```

File: src/style.css [hunk 20]
```diff
@@ -1,3 +1,139 @@
...
+}
+
+.card__buy:hover {
+  color: var(--button-color);
+  border: 1px solid var(--button-color);
```
Comment (Viktor-Kost) [inline src/style.css:137]:
```suggestion
  border-color: var(--button-color);
```

File: src/style.css [hunk 21]
```diff
@@ -1,3 +1,139 @@
...
+  font-weight: 700;
+  line-height: 18px;
+}
+
+.card__buy {
+  margin-top: 16px;
+  width: 100%;
+  height: 40px;
+  padding: 0;
+  color: white;
```
Comment (Viktor-Kost) [inline src/style.css:122]: it's better to use hex values instead of reserved words
```suggestion
  color: #ffffff;
```

File: src/style.css [hunk 22]
```diff
@@ -1,3 +1,125 @@
 body {
+  font-family: Roboto, Helvetica, sans-serif;
+  font-weight: 400;
+  --button-color: rgba(0, 172, 220, 1);
```
Comment (TarasHoliuk) [inline src/style.css:4]: not fixed:
image
rgba is redundant here - use rgb instead

File: src/style.css [hunk 23]
```diff
@@ -1,3 +1,125 @@
...
+  background-repeat: no-repeat;
+}
+
+.card__star:not(:last-child) {
+  margin-right: 4px;
+}
+
+.card__reviews-text {
+  font-size: 10px;
+  font-weight: 400;
```
Comment (TarasHoliuk) [inline src/style.css:81]: image
it means you can remove it

File: src/style.css [hunk 24]
```diff
@@ -1,3 +1,125 @@
...
+  font-family: Roboto, Helvetica, sans-serif;
+  font-weight: 400;
+  --button-color: rgba(0, 172, 220, 1);
+  --main-text-color: rgba(6, 11, 53, 1);
+  --secondary-text-color: rgba(97, 96, 112, 1);
+}
+
+* {
+  box-sizing: border-box;
   margin: 0;
```
Comment (TarasHoliuk) [inline src/style.css:11]: it's better to reset it for the list of tags you really need to avoid unexpected behavior

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]
Indeed - very good job, almost perfect :fire:
Let's add a few changes to make it perfect

File: src/style.css (hunk 24 above)
Comment (Cy90sure) [reply src/style.css:11]: I'm not sure if I need to do this because earlier I was told to _move box-sizing: border-box to * selector_...

Comment (denys-danyliuk) [reply src/style.css:11]: you can leave box-sizing here, but move margin to other selectors, it's inefficient to reset them under this selector

File: src/style.css [hunk 25]
```diff
@@ -1,3 +1,124 @@
...
+  --main-text-color: rgb(6, 11, 53);
+  --secondary-text-color: rgb(97, 96, 112);
+}
+
+* {
+  box-sizing: border-box;
   margin: 0;
 }
+
+.card {
```
Comment (denys-danyliuk) [inline src/style.css:14]: this file should be named after the block, also move all styles not related to the block from this file

File: src/style.css [hunk 26]
```diff
@@ -1,3 +1,124 @@
...
+  height: 16px;
+  width: 16px;
+  background-image: url(images/star.svg);
+  background-position: center;
+  background-repeat: no-repeat;
+}
+
+.card__star:not(:last-child) {
+  margin-right: 4px;
+}
```
Comment (denys-danyliuk) [inline src/style.css:77]: create a separate block for stars

File: src/style.css [hunk 27]
```diff
@@ -1,3 +1,124 @@
...
+  color: var(--secondary-text-color);
+  align-self: flex-start;
+}
+
+.card__reviews {
+  margin-top: 16px;
+  display: flex;
+  width: 100%;
+  justify-content: space-between;
+}
```
Comment (denys-danyliuk) [inline src/style.css:52]: use sass nesting

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]

File: src/style.css (hunk 25 above)
Comment (Cy90sure) [reply src/style.css:14]: What file are you talking about? What should it be named? I don't understand

**Pull request closed without merging**